File: backend/src/main/java/com/agrodairy/delivery/entity/Delivery.java
package com.agrodairy.delivery.entity;

import com.agrodairy.order.entity.Order;
import com.agrodairy.subscription.entity.Subscription;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "deliveries")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Delivery {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id")
    private Order order;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subscription_id")
    private Subscription subscription;

    @Column(name = "delivery_date", nullable = false)
    private LocalDate deliveryDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private DeliveryStatus status;

    @Column(name = "delivered_at")
    private Instant deliveredAt;

    private String notes;
}

File: frontend/src/types/delivery.ts
export type DeliveryStatus = 'PENDING' | 'OUT_FOR_DELIVERY' | 'DELIVERED' | 'FAILED'

export interface Delivery {
  id: string
  orderId: string | null
  subscriptionId: string | null
  deliveryDate: string
  status: DeliveryStatus
  deliveredAt: string | null
  notes: string | null
}

export interface UpdateDeliveryStatusRequest {
  status: DeliveryStatus
  notes?: string
}

export interface DeliverySummaryItem {
  productId: string
  name: string
  totalQuantity: number
}

export interface DeliverySummary {
  totalStops: number
  totalItemsByProduct: DeliverySummaryItem[]
}
